x-pack/plugins/security_solution/server/endpoint/services/actions/clients/crowdstrike/crowdstrike_actions_client.test.ts

```diff
@@ -75,6 +79,70 @@ describe('CrowdstrikeActionsClient class', () => {
     });
   });
 
+  it('should save response with error in case of actionResponse containing errors', async () => {
+    const doc = {
+      '@timestamp': '123',
+      EndpointActions: {
+        action_id: '123-345-567',
+        data: {
+          command: 'isolate',
+          comment: 'test comment',
+          hosts: {
+            '1-2-3': {
+              name: 'Crowdstrike-1460',
+            },
+          },
+        },
+        expiration: '123',
+        input_type: 'crowdstrike',
+        type: 'INPUT_ACTION',
+      },
+      agent: { id: ['1-2-3'] },
+      meta: {
+        hostName: 'Crowdstrike-1460',
+      },
+      user: { id: 'foo' },
+    };
+
+    const actionResponse = {
+      data: {
+        errors: [{ message: 'error message' }],
+      },
+    };
+
+    // @ts-expect-error Purposely testing a private method
+    await crowdstrikeActionsClient.completeCrowdstrikeAction(actionResponse, doc);
```

Comment:
(optional) why not just use the `isolate`/`release` public methods? It calls this one private method in its logic right?

Reply:
Also - FYI: hope you realize that the only reason this is working here is because it looks like the TS transpiler does not actually use the JS native private methods when it converts TS to JS.

Reply:
Thanks - I know it's not a good practice to unit test private methods, I try not to do that, but in this case it was easier to prepare the response. However, I agree with you comments so I tried harder, and came up with mocking the execute action - which results in calling isolate instead of completeAction. 

Thanks for the tip :)

Reply:
One way to test internal class methods is to make them `protected` instead and then expose them in the test by creating a sub-class. I did that with the SentinelOne response actions client class.